R/kernel.r

```diff
@@ -242,6 +243,75 @@ complete = function(request) {
         cursor_end = start_position + nchar(c.info$token)))
 },
 
+inspect = function(request) {
+    # 5.0 protocol:
+    code <- request$content$code
+    cursor_pos <- request$content$cursor_pos
+
+    # Define functions to add a section to content.
+    add_new_section_plain_text <- function (content, section_name, new_content) {
+        if (is.null(new_content)) return(content)
+        title <- paste0('# ', section_name, ':\n')
+        return(paste0(content, title, new_content, '\n', sep = '\n'))
+    }
+
+    add_new_section_html <- function (content, section_name, new_content) {
+        if (is.null(new_content)) return(content)
+        title <- paste0('<h1>', section_name, ':</h1>\n')
+        return(paste0(content, title, new_content, '\n', sep = '\n'))
+    }
+
+    new_section_adders <- list(
+        'text/plain' = add_new_section_plain_text,
+        'text/html' = add_new_section_html)
+
+    add_new_section <- function (data, section_name, new_data) {
+        for (mime in names(new_section_adders)) {
+            data[[mime]] <- new_section_adders[[mime]](
+                data[[mime]], section_name, new_data[[mime]])
+        }
+        return(data)
+    }
+
+    # Get token under the cursor_pos.
+    # There may be better ways to do that than this.
+    token <- ''
+    for (i in seq(cursor_pos, nchar(code))) {
+        token_candidate <- utils:::.guessTokenFromLine(code, i)
+        if (nchar(token_candidate) == 0) break
+        token <- token_candidate
+    }
+
+    data <- namedlist()
+    if (nchar(token) != 0) {
+        tryCatch(
+            {
+                obj <- eval(parse(text = token))
```

Comment:
For the cases such as `token` is `"1"` or `"NULL"`. We can't get the objects for them by `get(token)`, but we can by `eval(parse(text = token))`.
Given above, I think of continuing to use `eval()` and adding comments for the lines. Is it OK?